core/src/main/java/ai/timefold/solver/core/config/heuristic/selector/entity/EntitySelectorConfig.java

```diff
@@ -246,12 +269,23 @@ public void setSelectedCountLimit(@Nullable Long selectedCountLimit) {
         return this;
     }
 
+    /**
+     * @deprecated Deprecated in favor of {@link #withSorterComparatorFactoryClass(Class)}
+     * @param weightFactoryClass the factory class
+     */
+    @Deprecated(forRemoval = true, since = "1.28.0")
     public @NonNull EntitySelectorConfig
```

Comment:
Same comment on `@NonNull`. `@NullMarked` is a preferred alternative and if we're already touching the code, let's no longer introduce `@NonNull` if we can do `@NullMarked`.